src/main/java/com/hubspot/smtp/client/SmtpSessionConfig.java

```diff
@@ -2,24 +2,29 @@
 
 import java.net.InetSocketAddress;
 
-public class SmtpSessionConfig {
-  private final InetSocketAddress remoteAddress;
-  private final InetSocketAddress localAddress;
+import org.immutables.value.Value.Default;
+import org.immutables.value.Value.Immutable;
 
-  public static SmtpSessionConfig forRemoteAddress(InetSocketAddress remoteAddress) {
-    return new SmtpSessionConfig(remoteAddress, null);
+@Immutable
+public abstract class SmtpSessionConfig {
+  public abstract InetSocketAddress getRemoteAddress();
+  public abstract InetSocketAddress getLocalAddress();
+
+  @Default
+  public int getReadTimeoutSeconds() {
```

Comment:
jdk8 now has a `Duration` type you can probably use here.

`Duration.ofSeconds(30)` as a default.

Then in your code you'd call `getReadTimeout().get(ChronoUnit.MILLIS)`